docs/data_sharing_guide.md

```diff
@@ -0,0 +1,338 @@
+    table_name="climate_dataset"
+)
+
+print(f"Table is now private: {not response.is_public}")
+```
+
+## Managing Access Information
+
+### Check Who Has Access to Paths
+
+```python
+# Get detailed access information for any path that in your SQL warehouse
+access_info = get_path_access_info(
+    path="s3a://cdm-lake/users-general-warehouse/alice/datasets/climate-analysis"
+)
+
+print(f"Path: {access_info.path}")
+print(f"Users with access: {access_info.users}")
+print(f"Groups with access: {access_info.groups}")
+print(f"Public access: {access_info.public}")
+```
+
+### View Your Complete Workspace
+
+```python
+# Get comprehensive workspace information
+workspace = get_my_workspace()
+
+print(f"🏠 Your workspace: {workspace.username}")
+print(f"📁 Home directories: {len(workspace.home_paths)}")
+for path in workspace.home_paths:
+    print(f"   - {path}")
+
+print(f"👥 Group memberships: {len(workspace.groups)}")
+for group in workspace.groups:
+    print(f"   - {group}")
+
+print(f"🔓 Total accessible paths: {len(workspace.accessible_paths)}")
+
+# Show shared paths (paths not owned by you)
+shared_paths = [
+    path for path in workspace.accessible_paths 
+    if not any(path.startswith(home) for home in workspace.home_paths)
+]
+print(f"🤝 Shared with you: {len(shared_paths)}")
+for path in shared_paths[:5]:  # Show first 5
+    print(f"   - {path}")
+```
+
+## Working with Shared Tables
+
+### Accessing Shared Tables in Spark
+
+When others share tables with you, you can access them directly in your Spark sessions:
+
+```python
+# get_spark_session is auto-imported, no need to import
+# Create Spark session
+spark = get_spark_session()
+
+# Option 1: Access shared tables via SQL (if shared SQL warehouse)
+try:
+    shared_df = spark.sql("SELECT * FROM colleague_research.climate_data")
+    print(f"📊 Successfully loaded shared table with {shared_df.count()} rows")
+    shared_df.show(5)
+except Exception as e:
+    print(f"❌ Could not access shared table via SQL: {e}")
+
+# Option 2: Access via direct path (if you know the path)
+shared_table_path = "s3a://cdm-lake/users-sql-warehouse/colleague/research.db/climate_data"
+try:
+    shared_df = spark.read.format("delta").load(shared_table_path)
+    print(f"📊 Successfully loaded shared table with {shared_df.count()} rows")
+except Exception as e:
+    print(f"❌ Could not access shared table: {e}")
+    # Check access
+    access_info = get_path_access_info(shared_table_path)
+    print(f"Your access granted: {get_my_workspace().username in access_info.users}")
+```
+
+### Creating Tables from Shared Data
+
+```python
+# Copy shared data to your own tables
+# All governance functions are automatically available
+shared_df = spark.sql("SELECT * FROM colleague_research.raw_data")
+
+# Create your own analysis table
+analysis_df = shared_df.groupBy("category").count()
+
+# Save to your SQL warehouse
+analysis_df.write.format("delta").mode("overwrite").saveAsTable("my_analysis.category_counts")
```

Comment:
This is a reiteration of an earlier comment, but this doesn't seem like it should work - how does the system know how to read from colleague_research in their SQL warehouse but write to my SQL warehouse? It seems like this is operating on the same warehouse

Reply:
created ticket - https://kbase-jira.atlassian.net/jira/software/projects/CDM/boards/88/backlog?selectedIssue=CDM-315